src/mpi/request/test.c

```diff
@@ -161,12 +136,36 @@ int MPI_Test(MPI_Request * request, int *flag, MPI_Status * status)
     }
 #endif /* HAVE_ERROR_CHECKING */
 
+    /* If this is a null request handle, then return an empty status */
+    if (*request == MPI_REQUEST_NULL) {
+        MPIR_Status_set_empty(status);
+        *flag = TRUE;
+        goto fn_exit;
+    }
```

Comment:
Moving this outside the `MPIR_Test_impl` function means that every function that is calling it would need to explicitly check for this case.  In this patch, for example, `MPIR_Wait_impl` is not checking for this.

Reply:
`MPIR_Wait_impl` is changed in a different patch, the original code isn't consistent, but the final code I believe both were checking this outside.
https://github.com/pmodels/mpich/pull/2951/files#diff-2023d8a87eecda777a2ff89cc5f57272L163

Reply:
What do you mean by the original code not being consistent?  Your first patch is introducing a bug that didn't exist in the original code.  Each patch should be consistent.  Otherwise, there won't be any point of separating those patches.

Reply:
There is no bug, since MPI wait have always checked for this expept for the test check in test_impl and wait check outside originally

Reply:
There is no bug, since MPI wait have always checked for this expept for the test check in test_impl and wait check outside originally

Reply:
Oh, yes, you are right.  The check was redundant in `MPI_Wait`, so removing it from `MPIR_Test_impl` doesn't get rid of that functionality.  I withdraw this comment.